crates/database/src/quotes.rs

```diff
@@ -144,6 +268,64 @@ WHERE expiration_timestamp < $1
         .map(|_| ())
 }
 
+/// One row in the `quote_interactions` table.
+#[derive(Clone, Debug, PartialEq, sqlx::FromRow)]
+pub struct QuoteInteraction {
+    pub quote_id: QuoteId,
+    pub index: i32,
+    pub target: Address,
+    pub value: BigDecimal,
+    pub call_data: Vec<u8>,
+}
+
+/// Stores interactions provided by the solver for quote.
+pub async fn insert_quote_interaction(
+    ex: &mut PgConnection,
+    quote_interaction: &QuoteInteraction,
+) -> Result<(), sqlx::Error> {
+    const QUERY: &str = r#"
+INSERT INTO quote_interactions (
+    quote_id,
+    index,
+    target,
+    value,
+    call_data
+)
+VALUES ($1, $2, $3, $4, $5)
+    "#;
+    sqlx::query(QUERY)
+        .bind(quote_interaction.quote_id)
+        .bind(quote_interaction.index)
+        .bind(quote_interaction.target)
+        .bind(&quote_interaction.value)
+        .bind(&quote_interaction.call_data)
+        .execute(ex)
+        .await?;
+    Ok(())
+}
+
+pub async fn insert_quote_interactions(
+    ex: &mut PgTransaction<'_>,
+    quote_interactions: &[QuoteInteraction],
+) -> Result<(), sqlx::Error> {
+    for interaction in quote_interactions {
+        insert_quote_interaction(ex, interaction).await?;
+    }
+    Ok(())
+}
+
+pub async fn get_quote_interactions(
+    ex: &mut PgConnection,
+    quote_id: QuoteId,
+) -> Result<Vec<QuoteInteraction>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT *
+FROM quote_interactions
+WHERE quote_id = $1
+        "#;
+    sqlx::query_as(QUERY).bind(quote_id).fetch_all(ex).await
+}
```

Comment:
Is it used in tests only? Should be moved there then.

Reply:
Moved.